src/KPM.jl

```diff
@@ -24,9 +24,9 @@ function KPMBuilder(h, A, kets, order, bandrange)
     eA = eltype(eltype(A))
     mulist = zeros(promote_type(eh, eA), order + 1)
     bandbracket = bandbracketKPM(h, bandrange)
-    h´ = matrixKPM(h)
-    A´ = matrixKPM(A)
-    kets´ = Matrix(kets, h)
+    h´ = matrixKPM(h, SparseMatrixCSC{eh, Int64})
+    A´ = matrixKPM(A, SparseMatrixCSC{eh, Int64})
```

Comment:
I think what we want to do here is get a flattened version of bloch(h), right? I would then leave this as `MatrixKPM(h)` and replace `matrixtype` with `flatten` in `MatrixKPM`. Note that `bloch!(similarmatrix(h, flatten), h)` does what you want.